Source/engine/random.hpp

```diff
@@ -29,6 +29,19 @@ void SetRndSeed(uint32_t seed);
  */
 uint32_t GetLCGEngineState();
 
+/**
+ * @brief Advance the global RandomNumberEngine state by the specified number of rounds
+ *
+ * Only used to maintain vanilla compatibility until logic requiring reproducable random number generation is isolated.
+ * @param count How many values to discard
+ */
+void DiscardRandomValues(unsigned count);
```

Comment:
i guess we should markAdvanceRndSeed as nodiscard at this point

Reply:
Want me to do that in this PR? Easy enough to rewrite the second commit :D

Reply:
if you have time :)